kpatch-build/create-diff-object.c

```diff
@@ -200,8 +281,16 @@ static void kpatch_compare_correlated_section(struct section *sec)
 	    sec1->sh.sh_flags != sec2->sh.sh_flags ||
 	    sec1->sh.sh_addr != sec2->sh.sh_addr ||
 	    sec1->sh.sh_addralign != sec2->sh.sh_addralign ||
-	    sec1->sh.sh_entsize != sec2->sh.sh_entsize)
-		DIFF_FATAL("%s section header details differ", sec1->name);
+	    sec1->sh.sh_entsize != sec2->sh.sh_entsize) {
+		/*
+		 * .rodata section sometimes have sh_addr generated
+		 * and it might differ in the patched *.o file. Skip,
+		 * this case.
+		 */
+		if (!strncmp(sec1->name, ".rodata", 7) &&
+		    sec1->sh.sh_addr == sec2->sh.sh_addr)
+			DIFF_FATAL("%s section header details differ", sec1->name);
+	}
```

Comment:
Instead of doing this, it would be better to just remove the `sec1->sh.sh_addr != sec2->sh.sh_addr ||` line above.

It doesn't really make sense to compare the section addresses anyway, since those will change if a previous section changes, not the current section.  The only reason that worked on x86 is because the address is always zero for some reason.